File: crates/cpp/tests/native_strings/guest_imported_fns.cpp
#include "./cpp_host/the_world_cpp_native.h"
#include <iostream>

using namespace std;

void foo::foo::strings::A(std::string_view x)
{
    std::cout << x << std::endl;
}
wit::string foo::foo::strings::B()
{
    wit::string b = wit::string::from_view(std::string_view("hello B"));
    return b;
}
wit::string foo::foo::strings::C(std::string_view a, std::string_view b)
{
    std::cout << a << '|' << b << std::endl;
    wit::string c = wit::string::from_view(std::string_view("hello C"));
    return c;
}

File: crates/cpp/tests/native_strings/rust-host/rust_native_host_lib/src/guest_imported_fns.rs
pub mod foo {
    pub mod foo {
        pub mod strings {

            pub fn a(a: String) {
                println!("{a}");
            }
            pub fn b() -> String {
                let s = String::from("hello B");
                s
            }

            pub fn c(a: String, b: String) -> String {
                println!("guest imported: {} | {}", a, b);
                let s = String::from("hello C");
                s
            }
        }
    }
}
